src/WidgetBase.ts

```diff
@@ -385,6 +391,13 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 			const { children, properties = {} } = dNode;
 			const { key } = properties;
 
+			const propagatedPropertyNames = this.getDecorator('propagateProperty');
```

Comment:
was about to comment on this. I was indeed hoping we'd be able to do this down the standard decoration mechanism in `@afterRender` rather than adding additional code to WidgetBase